rasa/importers/multi_project.py

```diff
@@ -74,11 +74,18 @@ def _init_from_file(self, path: Text) -> None:
 
     def _init_from_dict(self, _dict: Dict[Text, Any], parent_directory: Text) -> None:
         imports = _dict.get("imports") or []
-        imports = {os.path.join(parent_directory, i) for i in imports}
+        imports = [os.path.join(parent_directory, i) for i in imports]
         # clean out relative paths
-        imports = {os.path.abspath(i) for i in imports}
-        import_candidates = [p for p in imports if not self._is_explicitly_imported(p)]
-        self._imports = self._imports.union(import_candidates)
+        imports = [os.path.abspath(i) for i in imports]
+
+        # remove duplication
+        import_candidates = []
+        for i in imports:
+            if i in import_candidates or self._is_explicitly_imported(i):
```

Comment:
saves us a line and removes the `continue` statement
```suggestion
            if i not in import_candidates or self._is_explicitly_imported(i):
                import_candidates.append(i)
```

Reply:
It needs more modification because of trailing or condition.

    if not (i in import_candidates or self._is_explicitly_imported(i)):

or

    if i not in import_candidates and not self._is_explicitly_imported(i):


but, both cases cause codeclimate error:

> Function _init_from_dict has a Cognitive Complexity of 6 (exceeds 5 allowed). Consider refactoring.

Personally, I  prefer the style you suggested.

Should we ignore this error and save a line?

Reply:
yep, codeclima is always a good indicator but no strict rule at the moment 👍

Reply:
ok, let's remove `continue`.